xs2a-adapter-service-api-adapter/src/main/java/de/adorsys/xs2a/adapter/http/ApacheHttpClient.java

```diff
@@ -120,12 +130,49 @@ public String content(Request.Builder requestBuilder) {
             InputStream content = entity != null ? entity.getContent() : EmptyInputStream.INSTANCE;
 
             T responseBody = responseHandler.apply(statusCode, content, responseHeaders);
+            logResponse(response, responseHeadersMap, responseBody);
             return new Response<>(statusCode, responseBody, responseHeaders);
         } catch (IOException e) {
             throw new UncheckedIOException(e);
         }
     }
 
+    private <T> void logResponse(
+        CloseableHttpResponse response,
+        Map<String, String> headers,
+        T responseBody
+    ) {
+        logger.debug("<-- {}", response.getStatusLine());
+        headers.keySet().forEach(
```

Comment:
Maybe it is more efficient to use `headers.entrySet()` instead of `headers.keySet()` to prevent the extra call to map later (I mean this call from line 147: `headers.get(key)`). What do you think?

The same for line 166.

Reply:
thanks, good point